src/port.ts

```diff
@@ -17,6 +18,7 @@ export class Port {
         this.id = id;
         this.value = value;
         this.type = Consts.DeviceType.UNKNOWN;
+        this.mode = null;
```

Comment:
Is `mode` also cleared on device reconnect? If not it could lead into one device "inheriting" the mode of the previous device if you don't use autoSubscribe (which will reset it to default for the device on attach).

Reply:
I did not because the mode would be overwritten by the next subscribe (either auto or not, you have to recall subscribe when device change).

However, that's a good point regarding consistency so I fixed it in 3bc8404d61761c5a72800f1f2703c75738c995ce . It can also be used to know if the devices already have a subscription.